.github/config/markdown-link-check.json

```diff
@@ -5,6 +5,7 @@
      { "pattern": "^http[s]*://localhost.*" }
     ,{ "pattern": "^https://.*\\.sigstore\\.dev.*" }
     ,{ "pattern": "^{{.*}}$" }
-    ,{ "pattern": "^https://github.com/**/compare/**" }
+    ,{ "pattern": "^https://github.com/.*/compare/.*" }
+    ,{ "pattern": "%7B%7B.*%7D%7D" }
```

Comment:
let it be known now and for future reference.

The above `"^{{.*}}$"` is actually not working, because the link being checked is URL encoded BEFORE checking. Meaning the actual pattern of the link that we'd need to ignore in case of our lovely templated links is this one.